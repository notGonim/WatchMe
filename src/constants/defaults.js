export { default as Jumbotron } from '../components/jumbotron'
export { default as Footer } from '../components/footer'
export { default as According } from '../components/according'
export { default as Form } from '../components/form'
export { default as Header } from '../components/header'
export { default as Feature } from '../components/feature'

