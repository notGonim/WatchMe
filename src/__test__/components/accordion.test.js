import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import faqsData from '../../fixtures/faqs.json'
import { According } from '../../constants/defaults';


describe('<According />', () => {

    it('renders the <According /> with populated data', () => {
        const { container, getByText } = render(
            <According>
                <According.Title>Frequently Asked Questions</According.Title>
                <According.Frame>
                    {faqsData.map((item) => (
                        <According.Item key={item.id}>
                            <According.Header>{item.header}</According.Header>
                            <According.Body>{item.body}</According.Body>
                        </According.Item>
                    ))}
                </According.Frame>
            </According>
        )
        expect(getByText('Frequently Asked Questions')).toBeTruthy()
        expect(getByText('Where can I watch?')).toBeTruthy()
        expect(container.firstChild).toMatchSnapshot()
    })



    it('open and close the  <According /> ', () => {
        const { container, queryByText } = render(
            <According>
                <According.Title>Frequently Asked Questions</According.Title>
                <According.Frame>
                    {faqsData.map((item) => (
                        <According.Item key={item.id}>
                            <According.Header>{item.header}</According.Header>
                            <According.Body>{item.body}</According.Body>
                        </According.Item>
                    ))}
                </According.Frame>
            </According>
        )
        const whatIsNetflixBodyText =
        "Netflix is a streaming service that offers a wide variety of award-winning TV programmes, films, anime, documentaries and more – on thousands of internet-connected devices. You can watch as much as you want, whenever you want, without a single advert – all for one low monthly price. There's always something new to discover, and new TV programmes and films are added every week!";
  
      expect(queryByText(whatIsNetflixBodyText)).toBeFalsy();
      fireEvent.click(queryByText('What is Netflix?'));
      expect(queryByText(whatIsNetflixBodyText)).toBeTruthy();
  
      fireEvent.click(queryByText('What is Netflix?'));
      expect(queryByText(whatIsNetflixBodyText)).toBeFalsy();
      expect(container.firstChild).toMatchSnapshot();
    });




});