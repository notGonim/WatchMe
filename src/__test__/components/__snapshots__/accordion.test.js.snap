// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<According /> open and close the  <According />  1`] = `
<div
  class="according__Container-ecn6mc-0 fmuCHV"
>
  <div
    class="according__Inner-ecn6mc-2 kWgEPB"
  >
    <h1
      class="according__Title-ecn6mc-3 jRqxXu"
    >
       
      Frequently Asked Questions
    </h1>
    <div
      class="according__Frame-ecn6mc-1 EiLCw"
    >
       
      <div
        class="according__Item-ecn6mc-4 hmikqE"
      >
         
        <div
          class="according__Header-ecn6mc-5 hlfRJg"
        >
          What is Netflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="according__Item-ecn6mc-4 hmikqE"
      >
         
        <div
          class="according__Header-ecn6mc-5 hlfRJg"
        >
          How much does Netflix cost?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="according__Item-ecn6mc-4 hmikqE"
      >
         
        <div
          class="according__Header-ecn6mc-5 hlfRJg"
        >
          Where can I watch?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="according__Item-ecn6mc-4 hmikqE"
      >
         
        <div
          class="according__Header-ecn6mc-5 hlfRJg"
        >
          How do I cancel?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="according__Item-ecn6mc-4 hmikqE"
      >
         
        <div
          class="according__Header-ecn6mc-5 hlfRJg"
        >
          What can I watch on Netflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<According /> renders the <According /> with populated data 1`] = `
<div
  class="according__Container-ecn6mc-0 fmuCHV"
>
  <div
    class="according__Inner-ecn6mc-2 kWgEPB"
  >
    <h1
      class="according__Title-ecn6mc-3 jRqxXu"
    >
       
      Frequently Asked Questions
    </h1>
    <div
      class="according__Frame-ecn6mc-1 EiLCw"
    >
       
      <div
        class="according__Item-ecn6mc-4 hmikqE"
      >
         
        <div
          class="according__Header-ecn6mc-5 hlfRJg"
        >
          What is Netflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="according__Item-ecn6mc-4 hmikqE"
      >
         
        <div
          class="according__Header-ecn6mc-5 hlfRJg"
        >
          How much does Netflix cost?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="according__Item-ecn6mc-4 hmikqE"
      >
         
        <div
          class="according__Header-ecn6mc-5 hlfRJg"
        >
          Where can I watch?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="according__Item-ecn6mc-4 hmikqE"
      >
         
        <div
          class="according__Header-ecn6mc-5 hlfRJg"
        >
          How do I cancel?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="according__Item-ecn6mc-4 hmikqE"
      >
         
        <div
          class="according__Header-ecn6mc-5 hlfRJg"
        >
          What can I watch on Netflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
